File: 0412-fizz-buzz/0412-fizz-buzz.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
char** fizzBuzz(int n, int* returnSize) {
    *returnSize = n;
    char** result = (char**)malloc(n * sizeof(char*));
    if (!result) {
        printf("Memory allocation failed\n");
        exit(1);
    }

    for (int i = 1; i <= n; i++) {
        if (i % 3 == 0 && i % 5 == 0) {
            result[i - 1] = strdup("FizzBuzz");
        } else if (i % 3 == 0) {
            result[i - 1] = strdup("Fizz");
        } else if (i % 5 == 0) {
            result[i - 1] = strdup("Buzz");
        } else {
            result[i - 1] = (char*)malloc(12 * sizeof(char));  
            sprintf(result[i - 1], "%d", i);
        }
    }

    return result;
}

File: 0412-fizz-buzz/0412-fizz-buzz.java
class Solution {
    public List<String> fizzBuzz(int n) {
         List<String> arr = new ArrayList<>();
         int i=1;
        while(i<=n)
        {
            if (i % 3 == 0 && i % 5 == 0) 
            {
                arr.add("FizzBuzz");
            }
             else if (i % 3 == 0) 
             {
                arr.add("Fizz");
            }
             else if (i % 5 == 0) 
             {
                arr.add("Buzz");
            } 
            else 
            {
                arr.add(String.valueOf(i));
            }
            i++;
        }

        return arr;
    }
}